src/backend/libc/io/types.rs

```diff
@@ -97,3 +112,30 @@ pub(crate) const STDIN_FILENO: c::c_int = c::STDIN_FILENO;
 pub(crate) const STDOUT_FILENO: c::c_int = c::STDOUT_FILENO;
 #[cfg(not(windows))]
 pub(crate) const STDERR_FILENO: c::c_int = c::STDERR_FILENO;
+
+/// A buffer type used with `vmsplice`.
+/// It is guaranteed to be ABI compatible with the iovec type on Unix platforms and WSABUF on Windows.
+/// Unlike `IoSlice` and `IoSliceMut` it is semantically like a raw pointer,
+/// and therefore can be shared or mutated as needed.
+#[cfg(any(target_os = "android", target_os = "linux"))]
+#[repr(transparent)]
+pub struct IoSliceRaw(c::iovec);
```

Comment:
`IoSlice` has a lifetime parameter, to prevent it from holding a dangling pointer. Would it make sense to add that to `IoSliceRaw` too, to prevent it from outliving the `buf` it's constructed from?